actionpack/test/dispatch/routing_test.rb

```diff
@@ -4037,6 +4037,7 @@ def draw(&block)
     routes = ActionDispatch::Routing::RouteSet.new
     routes.draw(&block)
     @app = self.class.build_app routes
+    @routes = routes
```

Comment:
This lets us use routing assertions directly without having to use `with_routing`.